docs/configuration.md

```diff
@@ -59,6 +59,8 @@ Relative paths specified within the config file are relative to the config file
 
 - <a name="deprecateTypingAliases"></a> **deprecateTypingAliases** [boolean]: PEP 585 indicates that aliases to types in standard collections that were introduced solely to support generics are deprecated as of Python 3.9. This switch controls whether these are treated as deprecated. This applies only when pythonVersion is 3.9 or newer. The default value for this setting is `false` but may be switched to `true` in the future.
 
+- <a name="identifyUnreachableCode"></a> **identifyUnreachableCode** [boolean]: If enabled, code that is determined to be unreachable by type analysis is reported using a tagged hint. This setting does not affect code that is determined to be unreachable regardless of type analysis. such code is always reported as such. This setting also has no effect when when using the command-line version of pyright because it never emits tagged hints for unreachable code.
```

Comment:
I find the name confusing. This really means unreachable by static type analysis, right? I think it needs some way to distinguish this from unreachable without type analysis.

`identifyTypeAnalysisUnreachableCode`? Maybe that's too long.

Reply:
I was also thinking there should be a different error message for this class of errors.

Reply:
> Maybe that's too long.

Yeah, I think "identifyTypeAnalysisUnreachableCode" is too long and cumbersome. IMO, the name "identifyUnreachableCode " plus the documentation make it clear enough.

> I was also thinking there should be a different error message for this class of errors.

These are never reported as errors, and they never show up in the "problems" section. Or are you talking about the "hint" message that appears when you hover over the gray text? Yeah, I think it makes sense to use different wording for the two cases. We could use a variant of the wording from @luabud's PR.

Reply:
Yeah the message that shows up in VS code. The warning text. Hopefully that being different will alleviate some of the concerns we've seen lately.

Reply:
I agree that this name is confusing. The main way that users complain about the current behavior is by saying, "You are reporting X as unreachable, but it isn't". So enabling this feature via "identifyUnreachableCode=true` seems weird from that perspective.

Also, even when this feature is disabled we'd be identifying some code as unreachable.

How about changing this from a bool setting to an enum?

Reply:
I don't think it's a good idea to introduce an enumerated setting. This would be inconsistent with the framework we've established for config settings. All current config settings are either diagnostic severities or booleans, and they have well-defined override semantics in the config file and in `# pyright` comments.

Also, "syntactic" isn't an accurate description because unreachability based on non-type information includes non-syntactic cues. Refer to the documentation for more details.

If users want to disable all tagged hints, they have the option of doing so.

Users who are complaining about this are not interested in type analysis. They have type checking disabled, so they're (somewhat) understandably confused when pylance is reporting something that is based on type analysis. From a type analysis perspective, this code is unreachable. To users of mypy, this would not be confusing at all.

Reply:
The name `enableReachabilityHints` is worth considering as a boolean. The verb "enable" is more consistent with existing settings than "identify". I'm somewhat reluctant to use the word "hints" here, but maybe that's OK.

Here are some other options to consider: "enableReachableCode", "enableCodeReachability", "enableUnreachableAnalysis", "enableUnreachableCodeAnalysis"? These aren't quite accurate because unreachable code analysis is always enabled; this switch simply determines whether we report the results of the analysis.

Here's another consideration. There has been [some demand](https://github.com/microsoft/pyright/issues/2672) for a way to have pyright emit a real diagnostic for unreachable code detected through type analysis. Mypy has an option like this. I've been reluctant to add this because it's an option that we couldn't enable by default even in strict mode, but I could see us adding it in the future if there's sufficient demand. If we do, it should have a name like `reportUnreachable` or `reportUnreachableCode`. It would be nice if the name we choose for this feature had a form that starts with "report".

Reply:
@luabud, any thoughts on this overall approach or on the name of the config setting?

Reply:
I actually agree `identifyUnreachableCode` should be fine given the documentation is clear, but IMO the least confusing ones are `enableReachabilityAnalysis` or even  `analyzeReachability`  for consistency with ` analyzeUnannotatedFunctions` (although to your point, it may be better to not be consistent in this case since the unreachable code analysis is actually happening)

Reply:
OK, thanks for all the feedback. I'm going to switch the name to "enableReachabilityAnalysis".

I'll close this PR and create a new one with all of the changes squashed into one.